hw/ip/csrng/data/csrng.hjson

```diff
@@ -207,6 +207,48 @@
         }
       ]
     },
+    { name: "INT_STATE_NUM",
+      desc: "Internal state number register",
+      swaccess: "rw",
+      hwaccess: "hro",
+      regwen: "REGEN",
+      fields: [
+        {
+            bits: "3:0",
+            name: "INT_STATE_NUM",
+            desc: '''
+                  Setting this field will set the number for which internal state can be
+                  selected for a read access. Up to 16 internal state values can be chosen
+                  from this register. The actual number of valid internal state fields
+                  is set by NHwApps plus 1 software app. For those selections that point
+                  to reserved locations (greater than NHwApps plus 1), the returned value
+                  will be zero.
+                  '''
+        },
+      ]
+    },
+    {
+      name: "INT_STATE_VAL",
+      desc: "Internal state read access register",
+      swaccess: "ro",
+      hwaccess: "hrw",
+      hwext: "true",
+      hwre: "true",
+      fields: [
+        { bits: "31:0",
+          name: "INT_STATE_VAL",
+          desc: '''
+                Reading this register will dump out the contents of the selected internal state field.
+                Since the internal state field is 448 bits wide, it will require 14 reads from this
+                register to gather the entire field. Once 14 reads have been done, the internal read
```

Comment:
it may make sense to reset the read pointer every time `INT_STATE_NUM` is written in order to give SW a way to reset this reading mechanism (otherwise it might get out of sync due to SW errors).

Reply:
Was not worried about the SW case, but determined that if a debugger was used, it would be easy to loose track of where the pointer was so added the write and clear pointer function.

Reply:
great, thanks!